src/receiver/middleware/check-signature.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+const crypto = require('crypto')
+
+module.exports = (secret, version = 'v0', onError) => {
```

Comment:
While having `v0` defaulted in the method signature doesn't hurt, is this necessary since you already have v0 as the default option in https://github.com/MissionsAI/slapp/pull/110/files#diff-addef27f2e22d1f6f2e0a998aa9e0e93R46 where the other defaults are set? Not sure of other patterns in slapp where you utilize signature defaults for options.

Reply:
Good point, it is inconsistent having a default value in the signature here.  I think I'll remove it, and have it throw an error in the immediate function call when creating the middleware function if no version is provided.

Reply:
I think I'm going to take back what I said 🙃, we aren't using defaults in signatures anywhere else currently, but I think having it here is worth it, even if it adds some duplication.  There are use-cases where someone will want to use this middleware explicitly (require it directly), and opt out of the way slapp wires it up by default (we actually do this in a few apps that are using Slapp).  Having it still defaulted in those cases feels valuable to me.

Reply:
Hmm, I'm a bit torn because this is one of those configuration values that largely comes from Slack and seems to have a higher potential of change than any of their other custom things. If there was a new standard version that Slack migrates to, Slapp maintainers would have to change `v0` in multiple places.

While I totally get how you would want to make it easier for external direct requires to not have to worry about something like versioning in the currently conceivable uses of the middleware, I would argue that if the dev is going to use it explicitly, then the dev should explicitly set the version that they're checking against. I would imagine a check after the `!secret` code that would throw an error if a secret is set but a version is not provided.

That being said, my general bias is always towards explicit intentional usage which makes it more difficult for devs to remember and a higher chance of tedium so...Totally get that I'm arguing on the side of more work.

Reply:
Ok, you talked me off the ledge.  I decided to throw during middleware creation here instead of at runtime in `onError` for 2 reasons.

1. If omitted here, it's a developer error, and should be prominent.
1. `onError` is optional, so want to avoid silent failures.